Assets/Scripts/Extensions/StringExtensions.cs

```diff
@@ -54,4 +55,25 @@ public static bool IsAlphabetical(this string self)
         return !string.IsNullOrWhiteSpace(self) 
             && Regex.IsMatch(self, alphabeticalPattern);
     }
+
+    public static string ReplaceTemplates(this string self)
+    {
+        MatchCollection matches = new Regex(UIConstants.TemplatePattern).Matches(self);
+        if (matches.Count <= 0)
+        {
+            return self;
+        }
+
+        foreach (Match match in matches.Cast<Match>().Reverse())

```

Comment:
why reverse?

Reply:
Otherwise the indexes for the later matches get shifted. As we are modifying the string. And the original match indexes will be different when it gets modified.